internal/aliasgen/aliasgen.go

```diff
@@ -0,0 +1,432 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by aliasgen. DO NOT EDIT.
+
+// Package %s aliases all exported identifiers in package
+// %q.
+//
+// Deprecated: Please use types in: %s
+package %s
+
+import (
+	src %q
+	grpc "google.golang.org/grpc" 
+)
+
+`, time.Now().Year(), am.pkg, am.importPath, am.importPath, am.pkg, am.importPath)
+	if _, err := io.Copy(w, strings.NewReader(header)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (am *aliasGenerator) writeConsts(w io.Writer) error {
+	if len(am.consts) == 0 {
+		return nil
+	}
+	if _, err := fmt.Fprintf(w, "// Deprecated: Please use consts in: %s\n", am.importPath); err != nil {
+		return err
+	}
+	if _, err := fmt.Fprintf(w, "const (\n"); err != nil {
+		return err
+	}
+	for _, v := range am.consts {
+		if _, err := fmt.Fprintf(w, "\t%s = src.%s\n", v, v); err != nil {
+			return err
+		}
+	}
+	if _, err := fmt.Fprintf(w, ")\n\n"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (am *aliasGenerator) writeVars(w io.Writer) error {
+	if len(am.vars) == 0 {
+		return nil
+	}
+	if _, err := fmt.Fprintf(w, "// Deprecated: Please use vars in: %s\n", am.importPath); err != nil {
+		return err
+	}
+	if _, err := fmt.Fprintf(w, "var (\n"); err != nil {
+		return err
+	}
+	for _, v := range am.vars {
+		if _, err := fmt.Fprintf(w, "\t%s = src.%s\n", v, v); err != nil {
+			return err
+		}
+	}
+	if _, err := fmt.Fprintf(w, ")\n\n"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (am *aliasGenerator) writeTypeNames(w io.Writer) error {
+	for _, v := range am.typeNames {
+		if v.doc != "" {
+			if _, err := fmt.Fprint(w, formatComment(v.doc, am.importPath)); err != nil {
+				return err
+			}
+		}
+		if _, err := fmt.Fprintf(w, "type %s = src.%s\n", v.name, v.name); err != nil {
+			return err
+		}
+	}
+	if _, err := fmt.Fprintf(w, "\n"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (am *aliasGenerator) writeFuncs(w io.Writer) error {
+	for _, f := range am.funcs {
+		if _, err := fmt.Fprintf(w, "// Deprecated: Please use funcs in: %s\n", am.importPath); err != nil {
+			return err
+		}
+		if _, err := fmt.Fprintf(w, "func %s(", f.name); err != nil {
+			return err
+		}
+
+		// write param info
+		for i, p := range f.params {
+			if i != 0 {
+				if _, err := fmt.Fprintf(w, ", "); err != nil {
+					return err
+				}
+			}
+			if _, err := fmt.Fprintf(w, "%s %s", p.name, p.FullType(am.pkg)); err != nil {
+				return err
+			}
+		}
+		if _, err := fmt.Fprintf(w, ")"); err != nil {
+			return err
+		}
+
+		// build return info
+		for i, r := range f.returns {
+			if i == 0 {
+				if _, err := fmt.Fprintf(w, " "); err != nil {
+					return err
+				}
+				if len(f.returns) > 1 {
+					if _, err := fmt.Fprintf(w, "("); err != nil {
```

Comment:
I believe this is untested, maybe add a func with multiple return values to `fakepb`?

Reply:
This actually never happens so I will just simpify code and return an error if it does.